File: Examples/Node/Tutorial/trivialHTTPServer.js
var fabric = require('Fabric');
var http = require('http');

http.createServer(function (req, res) {
  (function (fabricClient) {
    res.writeHead(200, {'Content-Type': 'text/plain'});
    res.end(fabricClient.build.getName() + " version " + fabricClient.build.getFullVersion() + "\n");
    fabricClient.close();
  })(fabric.createClient());
}).listen(1337, "127.0.0.1");

console.log('Server running at http://127.0.0.1:1337/');

File: Examples/Python/Tutorial/trivialHTTPServer.py
import fabric
import BaseHTTPServer

class FabricHandler( BaseHTTPServer.BaseHTTPRequestHandler ):
  def do_GET( self ):
    client = fabric.createClient()
    self.send_response( 200 )
    self.send_header( 'Content-type', 'text/plain' )
    self.end_headers()
    self.wfile.write( client.build.getName() + ' version ' + client.build.getFullVersion() + '\n' )
    client.close()

port = 1337
httpd = BaseHTTPServer.HTTPServer( ('', port), FabricHandler )
print('Server running at http://127.0.0.1:' + str(port) + '/')
httpd.serve_forever()

